File: OnlineStore/Api/CommonEntities/Additional/ProductStatus.cs
using System;


namespace CommonEntities.Additional
{
	public class ProductStatus
	{
		public const string StatusAvailable = "Available";
		public const string StatusNeedToOrder = "NeedToOrder";
		public const string StatusNotAvailable = "NotAvailable";
		
		public enum StatusEnum
		{
			Available = 10,
			NeedToOrder = 11,
			NotAvailable = 12
		}

		public StatusEnum Status { get; }


		public ProductStatus(string status)
		{
			Status = GetStatus(status);
		}
		public ProductStatus(StatusEnum status)
		{
			Status = status;
		}


		public string GetStatusName()
		{
			return GetStatusName(Status);
		}
		

		public static string GetStatusName(StatusEnum status)
		{
			switch (status)
			{
				case StatusEnum.Available: return StatusAvailable;
				case StatusEnum.NeedToOrder: return StatusNeedToOrder;
				case StatusEnum.NotAvailable: return StatusNotAvailable;
				default: throw new ArgumentOutOfRangeException(nameof(status), status, null);
			}
		}
		public static StatusEnum GetStatus(string status)
		{
			switch (status)
			{
				case StatusAvailable: return StatusEnum.Available;
				case StatusNeedToOrder: return StatusEnum.NeedToOrder;
				case StatusNotAvailable: return StatusEnum.NotAvailable;
				default: throw new ArgumentOutOfRangeException(nameof(status), status, null);
			}
		}

		
	}
}

File: OnlineStore/OnlineStoreUI/src/model/entities/apiEntities/additional/productStatus.ts
export class ProductStatus {
    status:ProductStatusEnum;

    constructor(status:ProductStatusEnum){
        this.status = status;
    }

    static getStatusPrint(status:ProductStatusEnum){

        if(status == ProductStatusEnum.Available)
            return 'Product available';
        if(status == ProductStatusEnum.NeedToOrder)
            return 'Product need to order';
        if(status == ProductStatusEnum.NotAvailable)
            return 'Product not available';
        
        return '';
    }

    getStatusPrint(){

        if(this.status == ProductStatusEnum.Available)
            return 'Product available';
        if(this.status == ProductStatusEnum.NeedToOrder)
            return 'Product need to order';
        if(this.status == ProductStatusEnum.NotAvailable)
            return 'Product not available';
        
        return '';
    }

    static getAllProdcutStatus():ProductStatus[]{

        let allStatus: ProductStatus[] = [
            new ProductStatus(ProductStatusEnum.Available),
            new ProductStatus(ProductStatusEnum.NeedToOrder),
            new ProductStatus(ProductStatusEnum.NotAvailable),
        ]

        return allStatus;
    }


}

export enum ProductStatusEnum {
    Available = 10,
	NeedToOrder = 11,
	NotAvailable = 12
}